kornia/augmentation/functional.py

```diff
@@ -0,0 +1,286 @@
+    See :class:`~kornia.augmentation.functional.apply_vflip` for details.
+    """
+    if isinstance(input, tuple):
+        batch_size = input[0].shape[0] if len(input[0].shape) == 4 else 1
+    else:
+        batch_size = input.shape[0] if len(input.shape) == 4 else 1
+    params = pg._random_prob_gen(batch_size, p=p)
+    return apply_vflip(input, params, return_transform)
+
+
+def color_jitter(input: torch.Tensor, brightness: FloatUnionType = 0.,
+                 contrast: FloatUnionType = 0., saturation: FloatUnionType = 0.,
+                 hue: FloatUnionType = 0., return_transform: bool = False) -> UnionType:
+    """Generate params and apply operation on input tensor.
+
+    See :class:`~kornia.augmentation.param_gen._random_color_jitter_gen` for details.
+    See :class:`~kornia.augmentation.functional.apply_color_jitter` for details.
+    """
+    if isinstance(input, tuple):
+        batch_size = input[0].shape[0] if len(input[0].shape) == 4 else 1
+    else:
+        batch_size = input.shape[0] if len(input.shape) == 4 else 1
+    params = pg._random_color_jitter_gen(batch_size, brightness, contrast, saturation, hue)
+    return apply_color_jitter(input, params, return_transform)
+
+
+def random_grayscale(input: torch.Tensor, p: float = 0.5, return_transform: bool = False):
+    """Generate params and apply operation on input tensor.
+
+    See :class:`~kornia.augmentation.param_gen._random_prob_gen` for details.
+    See :class:`~kornia.augmentation.functional.apply_grayscale` for details.
+    """
+    if isinstance(input, tuple):
+        batch_size = input[0].shape[0] if len(input[0].shape) == 4 else 1
+    else:
+        batch_size = input.shape[0] if len(input.shape) == 4 else 1
+    params = pg._random_prob_gen(batch_size, p=p)
+    return apply_grayscale(input, params, return_transform)
+
+
+def apply_hflip(input: torch.Tensor, params: Dict[str, torch.Tensor], return_transform: bool = False) -> UnionType:
+    r"""Apply Horizontally flip on a tensor image or a batch of tensor images with given random parameters.
+    Input should be a tensor of shape (H, W), (C, H, W) or a batch of tensors :math:`(*, C, H, W)`.
+
+    Args:
+        params (dict): A dict that must have {'batch_prob': torch.Tensor}. Can be generated from
+        kornia.augmentation.param_gen._random_prob_gen
+        return_transform (bool): if ``True`` return the matrix describing the transformation applied to each
+        input tensor.
+
+    Returns:
+        torch.Tensor: The horizontally flipped input
+        torch.Tensor: The applied transformation matrix :math: `(*, 3, 3)` if return_transform flag
+                      is set to ``True``
+    """
+
+    if not torch.is_tensor(input):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+
+    if len(input.shape) == 2:
+        input = input.unsqueeze(0)
+
+    if len(input.shape) == 3:
+        input = input.unsqueeze(0)
```

Comment:
```python
    if len(input.shape) in (2, 3,):
        input = input.unsqueeze(0)
```

Reply:
however not quite sure about adding dimensions since later we do: ` input.view((-1, (*input.shape[-3:])))`

Reply:
Here, I think it is better to remove the code below to be more clear that we will only accept (H, W), (C, H, W) and (*, C, H, W):
```python
    input = input.unsqueeze(0)
    input = input.view((-1, (*input.shape[-3:])))
```